tsl/src/remote/healthcheck.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+#include <postgres.h>
+#include <funcapi.h>
+#include <utils/builtins.h>
+#include <utils/lsyscache.h>
+#include <utils/syscache.h>
+
+#include "healthcheck.h"
+#include "data_node.h"
+#include "dist_commands.h"
+#include "dist_util.h"
+
+/*
+ * Functions and data structures for printing a health check result.
+ */
+enum Anum_show_conn
+{
+	Anum_health_node_name = 1,
+	Anum_health_healthy,
+	Anum_health_in_recovery,
+	Anum_health_error,
+	_Anum_health_max,
+};
+
+#define Natts_health (_Anum_health_max - 1)
+
+/*
+ * Health check result.
+ *
+ * Produces a tuple with four columns:
+ *
+ * 1. Node name (NULL for the access node itself)
+ *
+ * 2. A "healthy" boolean. An access node returns "false" in case a data node
+ * is not responding.
+ *
+ * 3. A boolean telling if the node is in recovery. Note that the "healthy"
+ * status will be "true" while the node is in recovery.
+ *
+ * 4. An optional error message explaining the reason for "healthy" being
+ * false.
+ */
+static HeapTuple
+create_health_check_tuple(const char *data_node, bool in_recovery, TupleDesc tupdesc)
+{
+	Datum values[Natts_health];
+	bool nulls[Natts_health];
+
+	memset(nulls, true, sizeof(nulls));
+
+	if (NULL != data_node)
+	{
+		NameData *node_name = palloc(sizeof(NameData));
+		namestrcpy(node_name, data_node);
+		values[AttrNumberGetAttrOffset(Anum_health_node_name)] = NameGetDatum(node_name);
+		nulls[AttrNumberGetAttrOffset(Anum_health_node_name)] = false;
+	}
+
+	values[AttrNumberGetAttrOffset(Anum_health_healthy)] = BoolGetDatum(true);
```

Comment:
So healthy column is reserved for the future use in such case? Like internal consistency check etc

Reply:
It would be `false` if, for example, the data node cannot establish a connection to the DN. But it is not possible to test now since it throws an error. It would be `true` if the node appears healthy, but, e.g., `in_recovery` is also `true`. We could, of course, simply use `false` in both cases, but I figured it might be useful information to also let the caller know that the node is recovering.

Reply:
Changed this so that healthy is `false` when `in_recovery` is true.